arch/risc-v/src/common/riscv_pmp.c

```diff
@@ -644,9 +644,19 @@ int riscv_check_pmp_access(uintptr_t attr, uintptr_t base, uintptr_t size)
           continue;
         }
 
-      /* Does this address range match ? */
-
-      if (base >= entry.base || end <= entry.end)
+      /* Does this address range match ? Take partial matches into account.
```

Comment:
should we merge into previous patch

Reply:
No problem, I preserved the local history but I can also squash the parts you mentioned.